pkg/services/chgm/chgm.go

```diff
@@ -376,31 +412,132 @@ func (c Client) RemoveCCAMLimitedSupport(externalID string) (bool, error) {
 	return c.DeleteCCAMLimitedSupportReason(c.cluster.ID())
 }
 
-// GetNodeCount returns the total number of all nodes that are supposed to be in the cluster
+// GetRunningNodesCount return the number of running nodes that are currently running in the cluster
+func (c Client) GetRunningNodesCount(infraID string) (*RunningNodesCount, error) {
+    instances, err := c.ListRunningInstances(infraID) 
+	if err != nil {
+		return nil, err
+	}
+
+    runningNodesCount := &RunningNodesCount{
+        Master: 0,
+        Infra: 0,
+        Worker: 0,
+    }
+
+    for _, instance := range instances {
+        for _, t := range instance.Tags {
+            if *t.Key == "Name" {
+                switch {
+                case strings.Contains(*t.Value, "master"):
+                    runningNodesCount.Master++
+                case strings.Contains(*t.Value, "infra"):
+                    runningNodesCount.Infra++
+                case strings.Contains(*t.Value, "worker"):
+                    runningNodesCount.Worker++
+                default:
+                    continue
+                }
+            }
+        }
+
+    }
+
+    return runningNodesCount, nil
+}
+
+// GetExpectedNodesCount returns the mininum number of nodes that are supposed to be in the cluster
 // We do not use nodes.GetTotal() here, because total seems to be always 0.
-func (c Client) GetNodeCount() int {
+func (c Client) GetExpectedNodesCount() (*ExpectedNodesCount, error) {
 	nodes, ok := c.cluster.GetNodes()
 	if !ok {
 		// We do not error out here, because we do not want to fail the whole run, because of one missing metric
```

Comment:
polish: i believe this comment is leftover and can be removed as you do error out here